### PR TITLE
Bug 1155542 - [Messages][New Gaia Architecture] Centralizing the global components/styling into a folder

### PR DESCRIPTION
**Pull request by steveck-chung**

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4cf81fb2ae46ec62cb69f59162d2028e4dfb0dc9

**Comment by azasypkin** [comment]
For the new location of ```desktop-only``` it looks good to me, but I believe you have to update ```build/build.js``` script as well.

For the ```global``` name - I don't have any preference yet, initially I was thinking about the following structure:

* build
* locales
* .......
* views
 * thread
    * js
    * styles
    * _index.html_
 * shared
   * js
    * styles
* layouts
  * small
     * _index.html_
     * js/_index.js_

But likely we'll need files from shared/global for layouts folder as well, so it would be better to have it upper in the hierarchy like you did. I also not sure about ```layouts``` folder, I took it from https://github.com/gmarty/photo-browse, maybe you have better ideas @steveck-chung @julienw ?

**Comment by azasypkin** [comment]
Some more thoughts:

In #29607, I moved link_helper and link_action_helper to thread view specific folder, looks like it's used only there. And "style/images/icons/delete.png" should probably end up in shared/global folder as it's used by thread and thread list.

Regarding name and location of global folder I don't have strong opinion right now, honestly it doesn't really matter, we always can change it :), but just one more idea to consider - how about ```views/shared or views/global or views/common``` (if we put main index.html ----> views/index.html, instead of "layouts/small/index.html" from my previous comment)?

**Comment by steveck-chung** [comment]
@azasypkin I think putting these resource under `views/global` make sense, but I'm not sure we will need `layouts/`. Maybe we just need a `views/small/` folder if we really need another document for other scenario?

**Comment by azasypkin** [comment]
> Maybe we just need a views/small/ folder if we really need another document for other scenario?

Yeah, I'd say we can even have main ```index.html``` directly in views folder, no need in ```layouts/small``` or even ```small``` for now

**Comment by julienw** [comment]
do you think we could rename "desktop-only" to "desktop-mocks" or even "desktop" ? not sure, also not necessary :)

I honestly prefer "shared" to "global"; we're used to that name, it has a easy-to-understand meaning.

About `layouts` directory, no opinion. We likely don't need it now...

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=5553697c7d43a25598ef9f51daa7d4c66559a452

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=5fca0572f43587cab57aaffe6a94a0eb8226fd06

**Comment by azasypkin** [comment]
> Also I don't think we should move everything but only views-related files. For example time_headers.js should move, but threads.js shouldn't. Tell me what you think."

@julienw  You mean that ```threads.js, drafts.js, message_manager.js, inter_instance_event_dispatcher.js, activity_handler.js (maybe?), settings.js(maybe?)``` will live somewhere in ```apps/sms/services```?

Also we'll likely have ```services/shared (with js only of course)``` as well.

I'm wondering whether we'll have js files shared between views and services (more like exception, utils or something similar). If so, I'm not sure where we can put it :)

**Comment by julienw** [comment]
Yeah that's what I mean.

It's not clear in my head if we'll have several "services" workers.. in that case I don't know if we need a services/shared.

About a general "utils" file, maybe these are utils that would be useful for all gaia too (so in /shared)? I'm thinking of a small promise-utils library (defer + async), don't know if there is anything else needed.

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=786eda65ea37e7456c021a8f19edcacf9ba2a641

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=6ab94c7b15d0b5bf61a8642d41361e4dc66a4730

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4b8769ea431b73fd127373c09788aacb8c6c47c5

File: apps/sms/desktop-mock/navigator_moz_mobilemessage.js [hunk 1]
```diff
@@ -61,7 +61,7 @@
     }
   };
 
-  getTestFile('/js/desktop-only/assets/kitten.jpg', function(testImageBlob) {
+  getTestFile('desktop-mock/assets/kitten.jpg', function(testImageBlob) {
```
Comment (azasypkin) [inline apps/sms/desktop-mock/navigator_moz_mobilemessage.js]: nit: maybe ```/desktop-mock``` just for consistency?

**Comment by azasypkin** [comment]
Please don't forget to update ```test/unit/media``` references in ```conversation_test.js``` (it's suspicious that it's not failing with old links :)), ```smil_test.js``` (Treeherder noticed that) and ```desktop-mock/navigator_moz_mobilemessage.js```

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e0cd774ed607fb2144dc1e6f9df8b787cca2106e

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=cf09a868a2053eebc6b5151b1fac0b0f11fb5386

File: apps/sms/desktop-mock/navigator_moz_mobilemessage.js [hunk 2]
```diff
@@ -19,6 +19,7 @@
   var requests = {};
   var now = Date.now();
   var ONE_DAY_TIME = 24 * 60 * 60 * 1000;
+  var mediafolder = '/views/shared/test/unit/media/';
```
Comment (julienw) [inline apps/sms/desktop-mock/navigator_moz_mobilemessage.js]: it's just a nit, so do what you want here: I usually prefer to remove the final `/` and put it where you use it -- I think it makes it easier to see it's a directory.

but do what you think.

File: apps/sms/index.html [hunk 3]
```diff
@@ -71,18 +71,18 @@
...
     <script defer src="shared/js/fb/fb_reader_utils.js"></script>
     <script defer src="shared/js/event_dispatcher.js"></script>
-    <script defer src="js/navigation.js"></script>
-    <script defer src="js/inter_instance_event_dispatcher.js"></script>
-    <script defer src="js/time_headers.js"></script>
-    <script defer src="js/contacts.js"></script>
-    <script defer src="js/drafts.js"></script>
-    <script defer src="js/threads.js"></script>
-    <script defer src="js/utils.js"></script>
+    <script defer src="views/shared/js/navigation.js"></script>
```
Comment (julienw) [inline apps/sms/index.html]: eventually navigation will really be split in 2 parts, but well, it can be wherever :)

File: apps/sms/index.html [hunk 4]
```diff
@@ -71,18 +71,18 @@
...
+    <script defer src="views/shared/js/inter_instance_event_dispatcher.js"></script>
+    <script defer src="views/shared/js/time_headers.js"></script>
+    <script defer src="views/shared/js/contacts.js"></script>
```
Comment (julienw) [inline apps/sms/index.html]: unsure for contacts -- it's mostly a service, and also we'll likely want to handle this differently when we'll have the DB... but mozContacts is not available in workers. We'll see later, let's put it there for now

File: apps/sms/views/conversation/test/unit/conversation_test.js [hunk 5]
```diff
@@ -143,6 +142,7 @@ suite('conversation.js >', function() {
     realMozL10n = navigator.mozL10n;
     navigator.mozL10n = MockL10n;
 
+    var mediafolder = '/views/shared/test/unit/media/';
```
Comment (julienw) [inline apps/sms/views/conversation/test/unit/conversation_test.js]: same nit than earlier

File: apps/sms/views/conversation/test/unit/attachment_renderer_test.js [hunk 6]
```diff
@@ -57,6 +57,7 @@ suite('AttachmentRenderer >', function() {
       type: 'image/jpeg'
     });
 
+    var mediaFolder = '/views/shared/test/unit/media/';
```
Comment (julienw) [inline apps/sms/views/conversation/test/unit/attachment_renderer_test.js]: same nit that above

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=79c8cb35ee883de76026539cd7cfd9a6497e9c57

**Pull request merged**